src/internal/storage/fileset/index/index_test.go

```diff
@@ -122,6 +127,135 @@ func TestMultiLevel(t *testing.T) {
 	Check(t, "abcdefgh")
 }
 
+func TestConcatFuzz(t *testing.T) {
+	ctx := context.Background()
+	db := dockertestenv.NewTestDB(t)
+	tr := track.NewTestTracker(t, db)
+	_, chunks := chunk.NewTestStorage(t, db, tr)
+	cache := NewCache(chunks, 5)
+	// Create ten file sets and concatenate them into one file set.
+	var topIdxs []*Index
+	var totalNumFiles int
+	for i := 0; i < 10; i++ {
+		iw := NewWriter(ctx, chunks, uuid.NewWithoutDashes())
+		numFiles := rand.Intn(100000)
```

Comment:
would it be worth recording the sequence of rand values so that in case of failure we can run forensics?

Reply:
I am planning on doing a larger set of changes in a separate PR for reporting the seeds of fuzz tests.